framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -275,73 +278,63 @@ private void refineArrayAtLengthAccess(FieldAccessNode arrayLengthNode, CFStore
-            } else {
-                List<Long> lengthValues = ValueAnnotatedTypeFactory.getIntValues(lengthAnno);
-                rolv =
-                        new RangeOrListOfValues(
-                                RangeOrListOfValues.convertLongsToInts(lengthValues));
-            }
-            AnnotationMirror newArrayAnno = rolv.createAnnotation(atypefactory);
-            AnnotationMirror oldArrayAnno =
+        if (oldArrayAnno == null) {
+            oldArrayAnno =
                     atypefactory.getAnnotationMirror(
-                            arrayLengthNode.getReceiver().getTree(), ArrayLen.class);
-
-            if (oldArrayAnno == null) {
-                oldArrayAnno =
-                        atypefactory.getAnnotationMirror(
-                                arrayLengthNode.getReceiver().getTree(), ArrayLenRange.class);
-            }
+                            arrayLengthNode.getReceiver().getTree(), ArrayLenRange.class);
+        }
 
-            AnnotationMirror combinedAnno;
-            // If the array doesn't have an @ArrayLen annotation, use the new annotation.
-            // If it does have an annotation, combine the facts known about the array
-            // with the facts known about its length using GLB.
-            if (oldArrayAnno == null) {
-                combinedAnno = newArrayAnno;
-            } else {
-                combinedAnno =
-                        atypefactory
-                                .getQualifierHierarchy()
-                                .greatestLowerBound(oldArrayAnno, newArrayAnno);
-            }
-            Receiver arrayRec =
-                    FlowExpressions.internalReprOf(
-                            analysis.getTypeFactory(), arrayLengthNode.getReceiver());
-            store.insertValue(arrayRec, combinedAnno);
+        AnnotationMirror combinedAnno;
+        // If the array doesn't have an @ArrayLen annotation, use the new annotation.
+        // If it does have an annotation, combine the facts known about the array
+        // with the facts known about its length using GLB.
+        if (oldArrayAnno == null) {
+            combinedAnno = newArrayAnno;
         } else {
-            // If the array's length is bottom, then this is dead code, so the array's type
-            // should also be bottom.
-            lengthAnno =
-                    AnnotationUtils.getAnnotationByClass(value.getAnnotations(), BottomVal.class);
-            if (lengthAnno != null) {
-                Receiver arrayRec =
-                        FlowExpressions.internalReprOf(
-                                analysis.getTypeFactory(), arrayLengthNode.getReceiver());
-                store.insertValue(arrayRec, lengthAnno);
-            }
+            combinedAnno =
+                    atypefactory
+                            .getQualifierHierarchy()
+                            .greatestLowerBound(oldArrayAnno, newArrayAnno);
```

Comment:
Yes, you're right about the test case.  GLB is correct.  This code is suppose to be refining the type, my concern is that it might refine it too far (to bottom).  But I can't make a test that isn't just dead code.  So I actually think this is ok.